### PR TITLE
flavour check for sharkey and iceshrimp

### PR DESCRIPTION
**Pull request by technicat**
For now, recognize iceshrimp as firefish, same as catodon (both firefish forks). Sharkey is a misskey fork but separate from firefish.

Split the trends feature into separate features for trending posts/tags/links. Firefish only supports trending posts, and sharkey only supports trending tags.

### CODE REVIEW

File: Sources/TootSDK/Models/Instance.swift [hunk 1]
```diff
@@ -219,9 +219,12 @@ extension Instance {
         if version.lowercased().contains("akkoma") {
             return .akkoma
         }
-        if version.lowercased().contains("firefish") {
+        if version.lowercased().contains("firefish") || version.lowercased().contains("iceshrimp") {
```
Comment (kkostov) [inline Sources/TootSDK/Models/Instance.swift:222]: With the additional logics, would it be possible to add some unit tests around the new flavour detections?

File: Sources/TootSDK/TootClient/TootClient+Trends.swift [hunk 2]
```diff
@@ -33,7 +33,7 @@ extension TootClient {
     ///   - offset: Skip the first n results.
     /// - Returns: Array of ``Post``.
     public func getTrendingPosts(limit: Int? = nil, offset: Int? = nil) async throws -> [Post] {
-        try requireFeature(.trends)
```
Comment (kkostov) [inline Sources/TootSDK/TootClient/TootClient+Trends.swift:36]: Maybe I'm not seeing it, why does the feature need to be renamed to support the new flavour?

**Comment by kkostov** [comment]
@technicat Love it, thank you for adding this. Would it be possible to add a couple of tests around the flavour detection so that we have concrete version string examples in the repo?

**Comment by technicat** [comment]
> @technicat Love it, thank you for adding this. Would it be possible to add a couple of tests around the flavour detection so that we have concrete version string examples in the repo?

Do you want to anonymize the domains in the same way as the existing friendica test? They also have real contact info: name, uri, and email. What do you want in place of that?

**Comment by kkostov** [comment]
> Do you want to anonymize the domains in the same way as the existing friendica test? They also have real contact info: name, uri, and email. What do you want in place of that?

Indeed, if committing new samples, please remove or replace identifying information with lorem ipsum or similar.

**Review by kkostov — APPROVED** [review]
🚀

**Pull request merged**